elastic/integrations
# [1password] Add user fields for actor_details, object_details, and aux_details.

## What does this PR do?


We've added new details fields to the data model that we want to consumed through this integration. An example of the original message would be:

```
                 {
			"uuid": "2CA5O6WU4VOPE225X64BMDJ6M3",
			"timestamp": "2023-06-30T18:02:23.273264Z",
			"actor_uuid": "PPQYFXGORNDIRHFJ43QNOEKZ6A",
			"actor_details": {
				"uuid": "PPQYFXGORNDIRHFJ43QNOEKZ6A",
				"name": "Insights Test",
				"email": "jin.dhaliwal+insightstest@agilebits.com"
			},
			"action": "trvlaway",
			"object_type": "user",
			"object_uuid": "X3TLF2JNIVDKJAXU67SNSHCFBU",
			"object_details": {
				"uuid": "X3TLF2JNIVDKJAXU67SNSHCFBU",
				"name": "Member 7",
				"email": "jin.dhaliwal+m7@agilebits.com"
			},
			"session": {
				"uuid": "RD4MLUQFIZDW5G6MUFFBIYUUBM",
				"login_time": "2023-06-30T18:01:42.300478Z",
				"device_uuid": "gdbhmdzsecieklx5wmgr7qy6oe",
				"ip": "172.19.0.1"
			}
		}
```

## Checklist


## Author's Checklist


- [x] Integration includes the new details fields in processed event data

## How to test this PR locally


Follow the testing steps outlined here: https://developer.1password.com/docs/events-api/reference/#post-apiv1auditevents

1. Get the elastic stack up and running. 
2. Follow the integration steps to set up the 1Password integration. (You will need a 1password business account)
  - You can create a test account on start.b5dev.com and use Stripe test card `4242424242424242` 
  - You'll need to change the URL in the integration setup to events.b5dev.com
3. On your 1Password account add some members/groups/vaults/items. This will generate events that the integration will capture. 
4. Confirm that the audit events dashboard contains those events and the new details fields are present. 

## Screenshots

![Screenshot 2023-07-06 at 12 40 22 PM](https://github.com/elastic/integrations/assets/1211388/59c04611-4c30-4f66-9c5c-64cc35e145d6)
![Screenshot 2023-07-06 at 12 40 03 PM](https://github.com/elastic/integrations/assets/1211388/7104dc89-a1f2-467d-838b-3198240cdfb5)
